packages/sdk/src/auth/policy.ts

```diff
@@ -0,0 +1,372 @@
+import { Client } from "@deco/sdk/storage";
+import { LRUCache } from "lru-cache";
+
+// Cache duration
+const TWO_MIN = 1000 * 60 * 2;
+
+// Base roles
+export const BASE_ROLES_ID = {
+  OWNER: 1,
+  PUBLISHER: 2,
+  COLLABORATOR: 3,
+  ADMIN: 4,
+};
+
+// Typed interfaces
+export interface Statement {
+  effect: "allow" | "deny";
+  resource: string;
+}
+
+export interface Policy {
+  id: number;
+  name: string;
+  team_id: number | null;
+  statements: Statement[];
+}
+
+export interface Role {
+  id: number;
+  name: string;
+  description?: string | null;
+  team_id: number | null;
+}
+
+export interface RoleWithPolicies extends Role {
+  policies: Policy[];
+}
+
+export interface MemberRole {
+  member_id: number;
+  role_id: number;
+  role?: Role;
+}
+
+export interface RoleUpdateParams {
+  roleId: number;
+  action: "grant" | "revoke";
+}
+
+/**
+ * PolicyClient - Singleton class for managing policy access
+ */
+export class PolicyClient {
+  private static instance: PolicyClient | null = null;
+  private db: Client | null = null;
+  private userPolicyCache: LRUCache<string, Policy[]>;
+  private teamRolesCache: LRUCache<number, Role[]>;
```

Comment:
can we use `@deco/sdk/cache` instead? so we can cache across workers

Reply:
a good approach would be:

1. cache for 1min~5min
2. delete before update (so we can keep it strongly consistent)